src/core/SpyRegistry.js

```diff
@@ -175,18 +175,34 @@ getJasmineRequireObj().SpyRegistry = function(j$) {
         prop,
         descriptor;
 
-      while (pointer) {
-        for (prop in pointer) {
-          if (
-            Object.prototype.hasOwnProperty.call(pointer, prop) &&
-            pointer[prop] instanceof Function
-          ) {
-            descriptor = Object.getOwnPropertyDescriptor(pointer, prop);
-            if (
-              (descriptor.writable || descriptor.set) &&
-              descriptor.configurable
-            ) {
-              props.push(prop);
+      while (
+        pointer &&
+        (!includeNonEnumerable || pointer !== Object.prototype)
+      ) {
+        var enumerableProperties = Object.keys(pointer);
+        var properties = includeNonEnumerable
```

Comment:
I find long multiline ternaries hard to read, especially when they have contain large expressions like the filter call here. I'd rather see this written as an if-else. There's some missing test coverage as well: if I make the filter function always return true, the tests still pass. I think there should be a test that verifies that non-enumerable constructors are filtered out and one that verifies that enumerable constructors are included.

Reply:
Removed the ternary, and added tests to make sure we are actually spying on enumberable constructors, but not non-enumerable constructors.